cmd/proxy/actions/app.go

```diff
@@ -75,12 +75,18 @@ func App(conf *config.Config) (*buffalo.App, error) {
 		app = app.Group(prefix)
 	}
 
-	// Register exporter to export traces
-	exporter, err := observ.RegisterTraceExporter(conf.TraceExporterURL, Service, ENV)
+	// RegisterExporter is used for registering and exporting traces from opencensus
+	// The error from the RegisterExporter would be nil if the tracer was specified by the user and
+	// and the trace exporter was created successfully.
+	// RegisterExporter returns a function that needs to be run for all services(datadog, jaeger)
+	// exporting traces from opencensus.
+	flushExporter, err := observ.RegisterExporter(conf.TraceExporter,
+		conf.TraceExporterURL,
+		Service, ENV)
 	if err != nil {
 		lggr.Infof("%s", err)
 	} else {
-		defer exporter.Flush()
+		defer flushExporter()
```

Comment:
Why did you choose to return the function flush instead of the exporter object.

The reason I am asking is because if you read the code, you are registering an exporter and would expect an exporter, not a function that flushes it and ends the object.

My major concerns are
A) To understand what flushExporter does, you will have to go and look into the package code
B) It is harder to extend the object in proxy app.

Reply:
@manugupt1 that was my suggestion. No need to create 1 interface and N structs. We can just return a `func()` that performs flushing. This PR will need a bunch of cleaning up to make it look a little bit more readable. Will add my comments as a review.

Reply:
Looking through your review, it makes sense!
Can you do the changes that @marwan-at-work  asked for?